proxy/src/proxy.rs

```diff
@@ -272,18 +277,24 @@ impl<S: AsyncRead + AsyncWrite + Unpin + Send> Client<'_, S> {
 
         let auth_result = async {
             // `&mut stream` doesn't let us merge those 2 lines.
-            let res = creds.authenticate(&extra, &mut stream).await;
+            let res = creds.authenticate(caches, &extra, &mut stream).await;
             async { res }.or_else(|e| stream.throw_error(e)).await
         }
         .instrument(info_span!("auth"))
         .await?;
 
-        let node = auth_result.value;
+        let mut node = auth_result.value;
         let (db, cancel_closure) = node
             .config
             .connect(params)
             .or_else(|e| stream.throw_error(e))
-            .await?;
+            .await
+            .map_err(|e| {
+                // Invalidate the cache entry if we failed to connect.
+                warn!("invalidating stalled compute node info cache entry");
```

Comment:
Is there enough context for this message. I find for non-ws connections `client=$session_id` but nothing for ws connections?

Reply:
I missed that there is the similar `ws-client=$session_id`, resolving.